costmap_converter/src/costmap_converter_node.cpp

```diff
@@ -125,20 +125,52 @@ class CostmapStandaloneConversion : public rclcpp::Node {
                               costmap_ros_->getCostmap(), true);
     }
 
+    last_publish_time_ = now();
+    cb_group1_ = this->create_callback_group(
+        rclcpp::CallbackGroupType::MutuallyExclusive);
+    cb_group2_ = this->create_callback_group(
+        rclcpp::CallbackGroupType::MutuallyExclusive);
     pub_timer_ = n_->create_wall_timer(
         std::chrono::milliseconds(200),
-        std::bind(&CostmapStandaloneConversion::publishCallback, this));
+        std::bind(&CostmapStandaloneConversion::publishCallback, this), cb_group1_);
+    health_check_timer_ = n_->create_wall_timer(
+        std::chrono::milliseconds(5000),
+        std::bind(&CostmapStandaloneConversion::healthCheck, this), cb_group2_);
   }
 
+  void healthCheck() {
```

Comment:
There is a chance that we are not able to respawn and that is if the issue is that the timers die. But if that happens, that's also good input, we'll just have to accept an operations intervention

Reply:
There is also a chance that this branch fixes the issue by mistake if e.g. the issue is in the SingleThreadedExecutor and we are having deadlocks